dali/pipeline/data/tensor_vector_test.cc

```diff
@@ -52,6 +59,633 @@ constexpr cudaStream_t cuda_stream = 0;
+  tv.set_order(order);
+  tv.SetContiguous(BatchState::Contiguous);
+  auto new_shape = TensorListShape<>{{1, 2, 3}, {2, 3, 4}, {3, 4, 5}};
+  tv.Resize(new_shape, DALI_FLOAT);
+  EXPECT_TRUE(tv.IsContiguous());
+
+  for (int i = 0; i < 3; i++) {
+    FillWithNumber(tv[i], 1 + i * 1.f);
+  }
+
+  for (int i = 0; i < 3; i++) {
+    tv.UnsafeCopySample(i, tv, i);
+  }
+
+  const auto *base = static_cast<const uint8_t*>(unsafe_sample_owner(tv, 0).get());
+
+  EXPECT_TRUE(tv.IsContiguous());
+  for (int i = 0; i < 3; i++) {
+    EXPECT_EQ(tv[i].raw_data(), base);
+    EXPECT_EQ(tv[i].shape(), new_shape[i]);
+    EXPECT_EQ(tv[i].type(), DALI_FLOAT);
+    base += sizeof(float) * new_shape[i].num_elements();
+    CompareWithNumber(tv[i], 1 + i * 1.f);
+  }
+
+  // Cannot copy without exact shape match when contiguous
+  EXPECT_THROW(tv.UnsafeCopySample(0, tv, 1), std::runtime_error);
+  EXPECT_THROW(tv.UnsafeCopySample(2, tv, 1), std::runtime_error);
+}
+
+
+TYPED_TEST(TensorVectorSuite, NoncontiguousResize) {
+  constexpr bool is_device = std::is_same_v<TypeParam, GPUBackend>;
+  const auto order = is_device ? AccessOrder(cuda_stream) : AccessOrder::host();
+  TensorVector<TypeParam> tv;
+  tv.set_order(order);
+  tv.SetContiguous(BatchState::Noncontiguous);
+  auto new_shape = TensorListShape<>{{1, 2, 3}, {2, 3, 4}, {3, 4, 5}};
+  tv.Resize(new_shape, DALI_FLOAT);
+  EXPECT_FALSE(tv.IsContiguous());
+
+  for (int i = 0; i < 3; i++) {
+    FillWithNumber(tv[i], 1 + i * 1.f);
+  }
+
+  for (int i = 0; i < 3; i++) {
+    EXPECT_NE(tv[i].raw_data(), nullptr);
+    EXPECT_EQ(tv[i].shape(), new_shape[i]);
+    EXPECT_EQ(tv[i].type(), DALI_FLOAT);
+    CompareWithNumber(tv[i], 1 + i * 1.f);
+  }
+
+  // Cannot copy without exact shape match when contiguous
+  tv.UnsafeCopySample(0, tv, 1);
+  tv.UnsafeCopySample(2, tv, 1);
+
+  EXPECT_FALSE(tv.IsContiguous());
+  for (int i = 0; i < 3; i++) {
+    EXPECT_NE(tv[i].raw_data(), nullptr);
+    EXPECT_EQ(tv[i].shape(), new_shape[1]);
+    EXPECT_EQ(tv[i].type(), DALI_FLOAT);
+    CompareWithNumber(tv[i], 2.f);
+  }
+}
+
+
+TYPED_TEST(TensorVectorSuite, BreakContiguity) {
+  TensorVector<TypeParam> tv;
+  // anything goes
+  tv.SetContiguous(BatchState::Default);
+
+  auto new_shape = TensorListShape<>{{1, 2, 3}, {2, 3, 4}, {3, 4, 5}};
+  tv.Resize(new_shape, DALI_FLOAT, BatchState::Contiguous);
+  EXPECT_TRUE(tv.IsContiguous());
+
+  for (int i = 0; i < 3; i++) {
+    FillWithNumber(tv[i], 1 + i * 1.f);
+  }
+
+  for (int i = 0; i < 3; i++) {
+    EXPECT_NE(tv[i].raw_data(), nullptr);
+    EXPECT_EQ(tv[i].shape(), new_shape[i]);
+    EXPECT_EQ(tv[i].type(), DALI_FLOAT);
+    CompareWithNumber(tv[i], 1 + i * 1.f);
+  }
+
+  Tensor<TypeParam> t;
+  t.Resize({2, 3, 4}, DALI_FLOAT);
+  SampleView<TypeParam> v{t.template mutable_data<float>(), t.shape(), DALI_FLOAT};
+  FillWithNumber(v, 42.f);
+
+  tv.UnsafeSetSample(1, t);
+  EXPECT_FALSE(tv.IsContiguous());
+
+  EXPECT_NE(tv[1].raw_data(), nullptr);
```

Comment:
why is the pointer nullptr?

Reply:
It is `NE`.

Reply:
I can probably compare with t.raw_data().

Reply:
misread, thanks.